Hublsoft.Net.LoginLogout.Api/Controllers/UserController.cs

```diff
@@ -18,18 +19,32 @@ namespace Hublsoft.Net.LoginLogout.Api.Controllers
     public class UserController : ControllerBase
     #pragma warning restore 1591
     {
+        private readonly IUserManager _userManager;
+
+        #pragma warning disable 1591
+        public UserController(IUserManager userManager)
+        #pragma warning restore 1591
+        {
+            _userManager = userManager;
+        }
+
         /// <summary>
         /// Retrieves the public id for the user given the request contains a valid api-key and a valid email address and password.
         /// </summary>
         /// <response code="200">Returns the public id for the user</response>
         /// <response code="400">If the request is malformed (email address is not supplied or is an empty string, password is not supplied or is an empty string, email address is not properly formatted)</response>
         /// <response code="401">If the request does not contain valid api-key</response>
-        /// <response code="403">The user could not be found (incorrect email address and password combination)</response>
+        /// <response code="403">The user could not be found, incorrect email address and password combination</response>
         /// <response code="500">An unhandled exception occurs</response>
         [HttpPost("authenticate")]
-        public IActionResult Authenticate(AuthenticateRequest model)
+        public async Task<ObjectResult> AuthenticateAsync(AuthenticateRequest model)
         {
-            return Ok(System.Guid.NewGuid());
+            var publicId = await _userManager.AuthenticateUserAsync(model.EmailAddress, model.Password);
+            if (publicId == Guid.Empty)
+            {
+                return new ForbiddenObjectResult("The user could not be found, incorrect email address and password combination");
+            }
+            return new OkObjectResult(publicId);
```

Comment:
ObjectResults allow you to pass an object in with the status code returned to the consumer. This greatly aids whoever is consuming the API by giving them a small amount of detail (e.g. in to why something failed). In case above, I will change the ForbiddenObjectResult to a NotFoundObjectResult. Also, perhaps naively believed, that ActionResult was/should be limited to returning response from an MVC controller.